datasource/timeouts/timeouts_test.go

```diff
@@ -0,0 +1,220 @@
+				"a": types.StringType,
+				"b": types.NumberType,
+				"c": types.BoolType,
+				"d": types.ListType{
+					ElemType: types.StringType,
+				},
+			}}},
+			expected: true,
+		},
+		"missing-attr": {
+			receiver: timeouts.Type{ObjectType: types.ObjectType{AttrTypes: map[string]attr.Type{
+				"a": types.StringType,
+				"b": types.NumberType,
+				"c": types.BoolType,
+				"d": types.ListType{
+					ElemType: types.StringType,
+				},
+			}}},
+			input: timeouts.Type{ObjectType: types.ObjectType{AttrTypes: map[string]attr.Type{
+				"a": types.StringType,
+				"b": types.NumberType,
+				"d": types.ListType{
+					ElemType: types.StringType,
+				},
+			}}},
+			expected: false,
+		},
+	}
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := test.receiver.Equal(test.input)
+			if test.expected != got {
+				t.Errorf("Expected %v, got %v", test.expected, got)
+			}
+		})
+	}
+}
+
+func TestTimeoutsValueRead(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		timeoutsValue   timeouts.Value
+		expectedTimeout time.Duration
+		expectedDiags   diag.Diagnostics
+	}
+	tests := map[string]testCase{
+		"read": {
+			timeoutsValue: timeouts.Value{
+				Object: types.ObjectValueMust(
+					map[string]attr.Type{
+						"read": types.StringType,
+					},
+					map[string]attr.Value{
+						"read": types.StringValue("10m"),
+					},
+				),
+			},
+			expectedTimeout: 10 * time.Minute,
+			expectedDiags:   nil,
+		},
+		"read-not-set": {
+			timeoutsValue: timeouts.Value{
+				Object: types.Object{},
+			},
+			expectedTimeout: 20 * time.Minute,
+			expectedDiags: diag.Diagnostics{
+				diag.NewErrorDiagnostic(
+					"Timeout Does Not Exist",
+					`timeout for "read" does not exist`,
+				),
+			},
+		},
+		"read-not-parseable-as-time-duration": {
+			timeoutsValue: timeouts.Value{
+				Object: types.ObjectValueMust(
+					map[string]attr.Type{
+						"read": types.StringType,
+					},
+					map[string]attr.Value{
+						"read": types.StringValue("10x"),
+					},
+				),
+			},
+			expectedTimeout: 20 * time.Minute,
+			expectedDiags: diag.Diagnostics{
+				diag.NewErrorDiagnostic(
+					"Timeout Cannot Be Parsed",
+					`timeout for "read" cannot be parsed, time: unknown unit "x" in duration "10x"`,
+				),
+			},
+		},
+	}
+
+	for name, test := range tests {
+		name, test := name, test
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			gotTimeout, gotErr := test.timeoutsValue.Read(context.Background())
+
+			if diff := cmp.Diff(gotTimeout, test.expectedTimeout); diff != "" {
+				t.Errorf("unexpected timeout difference: %s", diff)
+			}
+
+			if diff := cmp.Diff(gotErr, test.expectedDiags, equateErrorMessage); diff != "" {
+				t.Errorf("unexpected err difference: %s", diff)
+			}
+		})
+	}
+}
+
+// equateErrorMessage reports errors to be equal if both are nil
+// or both have the same message.
+var equateErrorMessage = cmp.Comparer(func(x, y error) bool {
```

Comment:
Is this necessary? I haven't come across needing anything special necessary with `go-cmp` comparing `diag.Diagnostics`, but if there is something special it'd be good to figure it out.

Reply:
Hmmmm..........have just checked and following the most recent code changes this no longer seems to be required.